independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ClientProxyGenerator.java

```diff
@@ -352,6 +362,22 @@ Collection<MethodInfo> getDelegatingMethods(BeanInfo bean, Consumer<BytecodeTran
         return methods.values();
     }
 
+    private void addDelegatesAndTrasformIfNecessary(Consumer<BytecodeTransformer> bytecodeTransformerConsumer,
+            boolean transformUnproxyableClasses,
+            Map<Methods.MethodKey, MethodInfo> methods, IndexView index,
+            Map<String, Set<Methods.NameAndDescriptor>> methodsFromWhichToRemoveFinal,
+            ClassInfo fieldClass) {
+        Methods.addDelegatingMethods(index, fieldClass, methods, methodsFromWhichToRemoveFinal,
+                transformUnproxyableClasses);
+        if (!methodsFromWhichToRemoveFinal.isEmpty()) {
```

Comment:
I wonder if we should only trigger transformations for normal scoped beans?

I.e. check the scope of the bean...

Reply:
Why do you think we should do that?

Reply:
Forget about it, we're talking about client proxies... so normal scoped beans only. My brain needs a restart.